src/main/java/Duke.java

```diff
@@ -1,10 +1,203 @@
+        }
+    }
+    public static void markTaskDone(ArrayList<Task> tasks, String command) /*throws DukeException*/ {
+
+        try {
+            int taskNumber = Integer.parseInt(command);
+            Task task = tasks.get(taskNumber - 1);
+            task.markAsDone();
+            printMarkDone();
+            printTask(task);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("The given task number does not exist. ):");
+            //throw in here
+        } catch (NumberFormatException e) {
+            System.out.println("The task index must be numeric.");
+        } catch (DukeException e) {
+            System.out.println("The task is already marked as done.");
+        }
+
+    }
+
+    public static void markTaskNotDone(ArrayList<Task> tasks, String command)  {
+
+        try {
+            int taskNumber = Integer.parseInt(command);
+            Task task = tasks.get(taskNumber - 1);
+            task.unmarkAsDone();
+
+            printMarkNotDone();
+            printTask(task);
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("The given task number does not exist. ):");
+            //throw in here
+        } catch (NumberFormatException e) {
+            System.out.println("The task index must be numeric.");
+        } catch (DukeException e) {
+            System.out.println("The task is already marked as not done.");
+        }
+    }
+    public static void addTodo(ArrayList<Task> tasks, String description) {
+        printAddTask();
+
+        Task task = new Todo(description);
+        tasks.add(task);
+
+        printTask(task);
+        printNoOfTasks(tasks.size());
+    }
+
+    public static void addDeadline(ArrayList<Task> tasks, String command) /*throws DukeException*/ {
+
+        if (command.contains("/by")) {
+            String[] components = command.split(" /by");
+
+            try {
+                Task task = new Deadline(components[0], components[1]);
+                tasks.add(task);
+                printAddTask();
+                printTask(task);
+                printNoOfTasks(tasks.size());
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("Please specify a time for the deadline.");
+            }
+
+        } else {
+            System.out.println("Invalid format. Remember to use '/by' to indicate the time.");
+            //throw new DukeException();
+        }
+    }
+
+    public static void addEvent(ArrayList<Task> tasks, String command) {
+
+        if (command.matches("(.*)" + "/from" + "(.*)" + "/to" + "(.*)")) {
+
+            try {
+                String[] components = command.split(" /from | /to "); //split string using "/from" and "/to"
+                Task task = new Event(components[0], components[1], components[2]);
+                tasks.add(task);
+
+                printAddTask();
+                printTask(task);
+                printNoOfTasks(tasks.size());
```

Comment:
Seems like these lines are constantly being repeated. Maybe you can try abstracting them to not repeat yourself